src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -191,6 +195,16 @@ def _accumulate_house_trades(house, grid, accumulated_trades, past_market_types)
                 seller_id = area_name_from_area_or_iaa_name(trade.seller)
                 accumulated_trades[house.name]["consumedFrom"][seller_id] += trade.offer.energy
                 accumulated_trades[house.name]["spentTo"][seller_id] += trade.offer.price
+
+    for market in getattr(house, past_market_types):
+        for trade in market.trades:
+            if area_name_from_area_or_iaa_name(trade.seller) == house.name:
+                accumulated_trades[house.name]["consumedFromExternal"] += trade.offer.energy
+                accumulated_trades[house.name]["spentToExternal"] += trade.offer.price
+            elif area_name_from_area_or_iaa_name(trade.buyer) == house.name:
+                accumulated_trades[house.name]["producedForExternal"] -= trade.offer.energy
+                accumulated_trades[house.name]["earnedFromExternal"] += trade.offer.price
```

Comment:
If we are following a sign convention. Then it has to be for both energy & price. For now, in this story it is irrelevant as you only need it for cumulative traded energy. But in case there is a need to plot cumulative energy bills, producer would be getting the money(to be shown in positive axis) whereas consumer would be paying the money(to be shown in negative axis).
So, I guess even now for uniformity to follow sign convention for both energy as well as for bills.

Reply:
You have a point, but @spyrostz was in favour of the += (see comment above).
I dont have a strong opinion on that. You are right, at the moment it is not used. So just let me know which one to implement. Thanks

Reply:
@faizan2590 the reason why the prices need to be absolute values, is that the purpose of this function is not to provide general-purpose statistics, but to provide an output for the frontend graph. I would agree with you if we wanted to store these accumulated statistics for further processing later, but since the purpose of this is only for the graph, I would suggest to have the aggregated statistics in the way that the graph expects them. 
The prices in this graph are only reported in the tooltip window (and I am not even sure if they are going to decide to display these) as absolute values. Therefore I would say to stick on unsigned values here.

Reply:
Ah sorry, didn't see @spyrostz  comment in your diff. Well then ignore it ATM. We will see later if there is some requirements later to plot "cumulative energy bills graph"

Reply:
Thank you @spyrostz this makes sense. Sticking with absolute values